test/intl402/NumberFormat/constructor-option-read-order.js

```diff
@@ -0,0 +1,36 @@
+/// Copyright 2023 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-initializenumberformat
+description: Checks the order of option read.
+features: [Intl.NumberFormat-v3]
+---*/
+
+let optionKeys =  Object.keys((new Intl.NumberFormat()).resolvedOptions());
+let opt = {};
+let readKeys = new Array();
+// For each item returned by resolvedOptions of default, add a getter
+// to track the reading order.
+optionKeys.forEach((property) =>
```

Comment:
I don't think I like this approach; `resolvedOptions()` properties are not guaranteed to correspond with properties _read_ during construction in either membership or sequence (and in fact the approach does fundamentally miss `localeMatcher` and incidentally also properties with no default—`currency`, `currencyDisplay`, `currencySign`, `unit`, `unitDisplay`, `minimumSignificantDigits`, `maximumSignificantDigits`, and `compactDisplay`). It would be better to be explicit, possibly with a check that fails in the presence of extra properties so we know to update the test.